src/overlays/actors/ovl_En_Zod/z_en_zod.c

```diff
@@ -28,62 +57,592 @@ ActorInit En_Zod_InitVars = {
+            this->stateFlags |= TIJO_STATE_2;
+            break;
+
+        default:
+            if (gSaveContext.save.weekEventReg[55] & 0x80) {
+                Actor_Kill(&this->actor);
+            }
+            this->actor.flags |= ACTOR_FLAG_10;
+            break;
+    }
+}
+
+void EnZod_Destroy(Actor* thisx, PlayState* play) {
+    EnZod* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void EnZod_HandleRoomConversation(EnZod* this, PlayState* play) {
+    u16 textId;
+
+    if (gSaveContext.save.playerForm != PLAYER_FORM_ZORA) {
+        textId = 0x1227;
+        if (gSaveContext.save.weekEventReg[32] & 8) {
+            textId = 0x1229;
+        } else {
+            gSaveContext.save.weekEventReg[32] |= 8;
+        }
+    } else if (this->stateFlags & TIJO_STATE_1) {
+        textId = 0x1225;
+    } else {
+        textId = 0x1219;
+        if (gSaveContext.save.weekEventReg[32] & 0x10) {
+            textId = 0x1226;
+        } else {
+            gSaveContext.save.weekEventReg[32] |= 0x10;
+        }
+        this->stateFlags |= TIJO_STATE_1;
+    }
+
+    EnZod_ChangeAnim(this, ENZOD_ANIM_PLAYING_VIVACE, ANIMMODE_ONCE);
+    Message_StartTextbox(play, textId, &this->actor);
+}
+
+s32 EnZod_PlayerIsFacingTijo(EnZod* this, PlayState* play) {
+    if ((this->actor.playerHeightRel < 30.0f) && (this->actor.xzDistToPlayer < 200.0f) &&
+        Player_IsFacingActor(&this->actor, 0x3000, play) && Actor_IsFacingPlayer(&this->actor, 0x3000)) {
+        return true;
+    }
+    return false;
+}
+
+void EnZod_ChangeAnim(EnZod* this, s16 nextAnimIndex, u8 mode) {
+    if ((nextAnimIndex < ENZOD_ANIM_PLAYING_VIVACE) || (nextAnimIndex >= ENZOD_ANIM_MAX)) {
+        nextAnimIndex = ENZOD_ANIM_PLAYING_LENTO;
+    }
+    Animation_Change(&this->skelAnime, sAnimations[nextAnimIndex], 1.0f, 0.0f,
+                     Animation_GetLastFrame(sAnimations[nextAnimIndex]), mode, -5.0f);
+    this->curAnimIndex = nextAnimIndex;
+    this->nextAnimIndex = nextAnimIndex;
+}
+
+void EnZod_UpdateAnimations(EnZod* this) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (this->nextAnimIndex == this->curAnimIndex) {
+            EnZod_ChangeAnim(this, this->curAnimIndex, ANIMMODE_ONCE);
+            switch (this->curAnimIndex) {
+                case ENZOD_ANIM_PLAYING_LENTO:
+                    if (Rand_ZeroFloat(1.0f) < 0.2f) {
+                        this->nextAnimIndex = ENZOD_ANIM_PLAYING_ANDANTINO;
+                    }
+                    break;
+
+                case ENZOD_ANIM_PLAYING_ANDANTINO:
+                    if (Rand_ZeroFloat(1.0f) < 0.8f) {
+                        this->nextAnimIndex = ENZOD_ANIM_PLAYING_LENTO;
+                    }
+                    break;
+            }
+
+        } else {
+            EnZod_ChangeAnim(this, this->nextAnimIndex, ANIMMODE_ONCE);
+        }
+        SkelAnime_Update(&this->skelAnime);
+    }
+}
+
+void EnZod_UpdateDrumVels(EnZod* this) {
```

Comment:
I don't like this name for two reasons: it updates the cymbals too, and it doesn't just update the velocities. `UpdateInstruments` is better overall IMO.